docs/RFCS/20220426_rbac_extensions_system_privileges.md

```diff
@@ -0,0 +1,137 @@
+
+# Technical Design
+Technical Design
+The goal here is to extend our current privilege system to allow more granular grants on objects / entities that are not necessarily backed by a descriptor such as jobs and changefeed objects.
+Certain role options should be replaced* with grants. At a high level, we want to introduce “system” or “global” level grants that follow general inheritance rules. This is a concept that exists in many popular non-Postgres databases. (MySQL, Oracle, SQL Server, Db2). We also want to support more granular grants on jobs and bulkio operations.
+
+## Replacing Relevant Role Options
+Note - to preface this, role options will likely not be deprecated, we can keep them side by side with their equivalent form of regular grants due to the overhead not being high enough to warrant a potentially tricky migration. We can emphasize in docs to use system privileges instead.
+
+In short, role options should be reserved for user administration permissions, anything that reads/writes from the database should be a grant.
+
+Role Option | Inheritable
+---- | ----
+CREATEROLE / NOCREATEROLE | Unchanged
+PASSWORD | Unchanged
+LOGIN / NOLOGIN | Unchanged
+VALIDUNTIL | Unchanged
+CONTROLJOB / NOCONTROLJOB | Replace with per job grants Example: GRANT PAUSE ON JOB TYPE BACKUP
+CONTROLCHANGEFEED / NOCONTROLCHANGEFEED | Changefeeds are created per table and write to a sink. We can support CHANGEFEED privilege on tables or on the database and also introduce grants on sinks
+CREATEDB / NOCREATEDB | Unchanged (Is a Postgres roleoption)
+CREATELOGIN / NOCREATELOGIN | Unchanged (?)
+VIEWACTIVITY / NOVIEWACTIVITY | Replace with system level grant
+CANCELQUERY / NOCANCELQUERY | Replace with system level grant
+MODIFYCLUSTERSETTING / NOMODIFYCLUSTERSETTING | Replace with system level grant
+VIEWACTIVITYREDACTED / NOVIEWACTIVITYREDACTED | Replace with system level grant
+SQLLOGIN / NOSQLLOGIN | Unchanged
+VIEWCLUSTERSETTING / NOVIEWCLUSTERSETTING | Replace with system level grant 
+
+##  New System Table
+
+We will create a new “system.privileges” table.
+ `
+CREATE TABLE system.privileges (
+username STRING NOT NULL,
+object INT NOT NULL,
+object_type system.object_type NOT NULL,
+privileges VARBIT NOT NULL,
+role_options VARBIT NOT NULL,
+CONSTRAINT "primary" PRIMARY KEY (username, object, object_type)
+);`
+
+`system.object_type` will be an enum that defines the various object types that can support privileges in CockroachDB. This tentatively looks like 
+```
+CREATE TYPE system.object_type AS ENUM('system', 'jobs', 'changefeed_sink')
```

Comment:
That's a fair concern, I think a valid question is how much value do we get out of this being an enum instead of just a string.

We can always do validation somewhere else in the code.